frontend/src/app/utils/timeline/timeline-creator.ts

```diff
@@ -293,4 +326,17 @@ export class TimelineCreator {
             throw new Error('Invalid scale');
         }
     }
+
+    private getBlockColor(taskCompletion: TaskCompletion) {
+        if (taskCompletion.isSuccessful) {
+            return 'green';
```

Comment:
Those should be constants.
Also, you should not use styles somewhere in unknow service. To be more correct, you should have some enum in the model that reflect the state. In your html you should use *ngClass and add correct color by class according to your state: e.g. *ngClass="{ 'completed-task' : model.state === CompletionsStates.Success }".